src/jit/morph.cpp

```diff
@@ -2734,29 +2747,22 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
 
         /* We must fill in or update the argInfo table */
 
-        if  (!lateArgsComputed)
+        if (lateArgsComputed)
+        {
+            /* this is a register argument - possibly update it in the table */
+            call->fgArgInfo->RemorphRegArg(argIndex, argx, NULL, genMapIntRegArgNumToRegNum(intArgRegNum), 1, 1);
+        }
+        else
         {
             assert(varTypeIsGC(call->gtCallObjp->gtType) ||
                    (call->gtCallObjp->gtType == TYP_I_IMPL));
 
             /* this is a register argument - put it in the table */
-            call->fgArgInfo->AddRegArg(argIndex,
-                                       argx, 
-                                       NULL, 
-                                       genMapIntRegArgNumToRegNum(intArgRegNum),
-                                       1, 
-                                       1
+            call->fgArgInfo->AddRegArg(argIndex, argx, NULL, genMapIntRegArgNumToRegNum(intArgRegNum), 1, 1
 #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
-                                       , false,
-                                       REG_STK,
-                                       nullptr
+                                       , false, REG_STK, nullptr
```

Comment:
I agree; that could be a follow-up cleanup (for someone, not necessarily me :-))